substrate/pkg/utils/kubectl/apply.go

```diff
@@ -0,0 +1,61 @@
+package kubectl
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/logging"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (c *Client) Apply(ctx context.Context, file string) error {
+	resp, err := http.Get(file)
+	if err != nil {
+		return fmt.Errorf("getting response, %w", err)
+	}
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("parsing response, %w", err)
+	}
+	if err := c.ApplyYAML(ctx, data); err != nil {
+		return fmt.Errorf("applying yaml, %w", err)
+	}
+	logging.FromContext(ctx).Infof("Applied %s", file)
+	return nil
+}
+
+func (c *Client) ApplyYAML(ctx context.Context, yamlBytes []byte) error {
+	objects, err := c.ParseObjects(ctx, yamlBytes)
```

Comment:
Should we have `ParseObjects` in the same file?

Reply:
This is generic to create/delete/etc, so I put it in the client.

Reply:
This is now refactored w/ dynamicclient.